### PR TITLE
SOP for deploying to beta.gigadb.org

### PR DESCRIPTION
**Pull request by pli888**
# Pull request for issue: #1013

This is a pull request for the following functionalities:

* An SOP for deploying to beta.gigadb.org

## Changes to the documentation

See https://github.com/pli888/gigadb-website/blob/beta-deploy-sop/docs/sop/BETA_DEPLOY.md

### CODE REVIEW

**Review by rija — COMMENTED** [review]
Hi @pli888,

I've started following the SOP instructions.

But before that I've created a new macOS user on my laptop dedicated to that.
This is because I don't trust myself not making mistakes when manipulating configs and credentials when switching back and forth between developer mode and upstream deploy mode.

It also helps with the reviewing (the two problems mentioned below would not have been found had I have performed the SOP in my regular user account).

With further setup which  I didn't bother to do for now,  it could also improve security.

So following the steps, I encountered two issues in the process of provisioning staging.gigadb.org:


```
gigascience@Kiteman staging % ../../../scripts/ansible_init.sh --env staging
{"variable_type":"env_var","key":"gigadb_db_host","value":"rds-server-staging-gigadb.cfkc0cbc20ii.ap-east-1.rds.amazonaws.com","protected":false,"masked":false,"environment_scope":"staging"}do_known_hosts: hostkeys_foreach failed: No such file or directory
```

this is because ``.ssh/known_hosts`` didn't exist. 
I had to create it with 

```
$ touch ~/.ssh/known_hosts
```

```
gigascience@Kiteman staging % TF_KEY_NAME=private_ip ansible-playbook -i ../../inventories webapp_playbook.yml
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details
ERROR! the role 'role-secure-docker-daemon' was not found in /Users/gigascience/gigadb-website/ops/infrastructure/envs/staging/roles:/Users/gigascience/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/gigascience/gigadb-website/ops/infrastructure/envs/staging

The error appears to be in '/Users/gigascience/gigadb-website/ops/infrastructure/envs/staging/webapp_playbook.yml': line 26, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

    - role: ../../roles/docker-install
    - role: role-secure-docker-daemon
      ^ here
```

This is because the third party Ansible roles were not installed. It was fixed with:

```
$ ansible-galaxy install -r ../../../infrastructure/requirements.yml
```


I'll continue tomorrow with the rest of the steps.

File: docs/sop/BETA_DEPLOY.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+Enable cronjob for periodically resetting the database:
+```
+$ ansible-playbook -i ../../inventories bastion_playbook.yml -e "reset_database_cronjob_state=present"
+```
+
+To confirm the cron job has been created, log in with ssh on bastion, and run 
+`crontab -l` - there should be a cronjob to reset the database daily at 10:05am 
+UTC (after the CNGB backup of day before is made available).
+
```
Comment (rija) [inline docs/sop/BETA_DEPLOY.md:301]: Probably a good idea to give an example here (that we can copy/paste), e.g: 
```
# get public ip of bastion server
$ terraform output

## ssh into it
$ ssh -i ~/.ssh/id-rsa-aws-hk-gigadb.pem centos@<bastion public ip>
```

**Comment by rija** [comment]
Hi @pli888,

I've provisioned both staging and live on upstream.
No major problems to report, I've added review comments for small things I've noted.

I was about to trigger Gitlab pipeline job but it is not working as for some reasons  I'm not authorized (despite being the owner of the group)

I'll look further into what's going on.

Screenshot 2022-04-28 at 4 23 23 PM

**Comment by rija** [comment]
> I was about to trigger Gitlab pipeline job but it is not working as for some reasons I'm not authorized (despite being the owner of the group)

Thanks @pli888, your suggestion was spot on and unblocked me. 
It's probably a good idea to add your gitter comment to the SOP too.

**Comment by rija** [comment]
Hi @pli888,

I have now successfully deployed both staging.gigadb.org and beta.gigadb.org

I have no further comments to add beyond the ones above.

**Comment by pli888** [comment]
>I have no further comments to add beyond the ones above.

@rija I've updated the SOP with your suggestions.

**Review by rija — APPROVED** [review]
Hi @pli888,

It all looks good, happy to approve it

**Pull request merged**